src/pages/workspaces/workspace/analysis/AzureComputeModal.js

```diff
@@ -0,0 +1,332 @@
+                isSearchable: false,
+                isClearable: false,
+                value: computeConfig.machineType,
+                onChange: ({ value }) => updateComputeConfig('machineType', value),
+                options: _.keys(azureMachineTypes),
+                getOptionLabel: ({ value }) => getMachineTypeLabel(value),
+                styles: { width: '400' }
+              })
+            ])
+          ])
+        ]),
+        div({ style: { display: 'flex', marginTop: '.5rem' } }, [
+          h(Link, { href: 'https://azure.microsoft.com/en-us/pricing/details/virtual-machines/series/', ...Utils.newTabLinkProps }, [
+            'Learn more about cloud compute profiles',
+            icon('pop-out', { size: 12, style: { marginLeft: '0.25rem' } })
+          ])
+        ])
+      ]),
+      div({ style: { marginBottom: '2rem' } }, [
+        h(IdContainer, [
+          id => h(Fragment, [
+            div({ style: { marginBottom: '.5rem' } }, [
+              label({ htmlFor: id, style: computeStyles.label }, ['Location'])
+            ]),
+            div({ style: { width: 400 } }, [
+              h(Select, {
+                id,
+                isSearchable: false,
+                isClearable: false,
+                value: computeConfig.region,
+                isDisabled: true, //this is currently locked to workspace location
+                onChange: ({ value }) => updateComputeConfig('region', value),
+                options: _.keys(azureRegions),
+                getOptionLabel: ({ value }) => getRegionLabel(value)
+              })
+            ])
+          ])
+        ])
+      ]),
+      div({ style: { marginBottom: '2rem' } }, [
+        h(IdContainer, [
+          id => h(Fragment, [
+            div({ style: { marginBottom: '.5rem' } }, [
+              label({ htmlFor: id, style: computeStyles.label }, ['Disk Size (GB)'])
+            ]),
+            div({ style: { width: 75 } }, [
+              h(NumberInput, {
+                id,
+                min: 50,
+                max: 64000,
+                isClearable: false,
+                onlyInteger: true,
+                value: computeConfig.diskSize,
+                onChange: v => updateComputeConfig('diskSize', v)
+              })
+            ])
+          ])
+        ])
+      ])
+    ])
+  }
+
+  // Will be used once we support update
+  // const hasChanges = () => {
+  //   const existingConfig = adaptRuntimeDetailsToFormConfig()
+  //
+  //   return !_.isEqual(existingConfig, computeConfig)
+  // }
+  //
+  // const adaptRuntimeDetailsToFormConfig = () => {
+  //   return currentRuntimeDetails ? {
+  //     machineType: currentRuntimeDetails.runtimeConfig?.machineType || defaultAzureMachineType,
+  //     diskSize: currentRuntimeDetails.diskConfig?.size || defaultAzureDiskSize,
+  //     region: currentRuntimeDetails.runtimeConfig?.region || defaultAzureRegion
+  //   } : {}
+  // }
+
+  const doesRuntimeExist = () => !!currentRuntimeDetails
```

Comment:
I like utilities like this that makes something potentially ambiguous into something explicit.